pkg/dynamicinstrumentation/testutil/e2e_test.go

```diff
@@ -0,0 +1,188 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf && arm64
+
+package testutil
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"reflect"
+	"strings"
+	"testing"
+	"text/template"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/dynamicinstrumentation"
+	"github.com/DataDog/datadog-agent/pkg/dynamicinstrumentation/diconfig"
+	"github.com/DataDog/datadog-agent/pkg/dynamicinstrumentation/ditypes"
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/features"
+	"github.com/cilium/ebpf/rlimit"
+	"github.com/kr/pretty"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestGoDI(t *testing.T) {
+	if err := rlimit.RemoveMemlock(); err != nil {
```

Comment:
If the test will fail without this, I'd suggest `require.NoError(t, rlimit.RemoveMemlock())`

Reply:
You are calling it twice now? You just need the single line `require.NoError(t, rlimit.RemoveMemlock())`